engineapi/rollout.go

```diff
@@ -0,0 +1,98 @@
+package engineapi
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	bsRollout "github.com/longhorn/backupstore/rollout"
+
+	"github.com/longhorn/longhorn-manager/types"
+
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+)
+
+// RolloutUpload uploads rollout file to the backup target
+func (btc *BackupTargetClient) RolloutUpload(name, localFile, longhornVersion, longhornGitCommit, managerImage, engineImage string) (string, error) {
+	output, err := btc.ExecuteEngineBinary(
+		"rollout", "upload",
+		"--source", localFile,
+		"--dest", btc.URL,
+		"--longhorn-version", longhornVersion,
+		"--longhorn-git-commit", longhornGitCommit,
+		"--rollout-name", name,
+		"--manager-image", managerImage,
+		"--engine-image", engineImage,
+	)
+	if err != nil {
+		return "", errors.Wrapf(err, "error uploading rollout %v", name)
+	}
+	return output, nil
+}
+
+// RolloutDownload downloads rollout from the backup target
+func (btc *BackupTargetClient) RolloutDownload(name, version, downloadPath string) error {
+	_, err := btc.ExecuteEngineBinary(
+		"rollout", "download",
+		"--source", btc.URL,
+		"--dest", downloadPath,
+		"--longhorn-version", version,
+		"--rollout-name", name,
```

Comment:
Do you mean if we can execute engine binary, and this command can get the detail from BackupTarget CR and RolloutSave CR instead of inputting them as command flags?

Reply:
I think I misunderstood this `"--rollout-name"` when adding this comment. Sorry for that. This cmd should be fine.

But I still prefer to rename the remote file from `rollout` to `rolloutsave`

Reply:
@innobead , WDYT? I think you also have some thoughts on the naming for `rollout-save`, `rollout`, and `rollout-load`.

We need to work out some names we agree on, then I will change them at once.

Reply:
Discussion ref: https://github.com/longhorn/longhorn/pull/4388#discussion_r997634359

Reply:
Updated.